app/services/authorisation.rb

```diff
@@ -0,0 +1,59 @@
+class Authorisation
+  class ExternalServerError < StandardError; end
+
+  attr_accessor :organisation_id,
+                :user_id,
+                :role_ids,
+                :dfe_sign_in_url,
+                :dfe_sign_in_password,
+                :dfe_sign_in_service_id,
+                :dfe_sign_in_service_access_role_id
+
+  def initialize(organisation_id:, user_id:)
+    self.organisation_id = organisation_id
+    self.user_id = user_id
+    self.dfe_sign_in_url = DFE_SIGN_IN_URL
+    self.dfe_sign_in_password = DFE_SIGN_IN_PASSWORD
+    self.dfe_sign_in_service_id = DFE_SIGN_IN_SERVICE_ID
+    self.dfe_sign_in_service_access_role_id = DFE_SIGN_IN_SERVICE_ACCESS_ROLE_ID
+  end
+
+  def call
```

Comment:
Thought about stripping this method down into more methods but felt that it is still quite easy to follow along as we build the request, execute it and handle the response. If this got any bigger it would be clearer to start chopping it up. Eager to hear another opinion.

Reply:
I'd probably add the response building (so lines 22 - 31) into another method, so something like:

```ruby
def build_response
  uri = URI(dfe_sign_in_url)
  uri.path = "/services/#{dfe_sign_in_service_id}/organisations/#{organisation_id}/users/#{user_id}"

  request = Net::HTTP::Get.new(uri)
  request['Authorization'] = "bearer #{generate_jwt_token}"
  request['Content-Type'] = 'application/json'

  Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
    http.request(request)
  end
end
```

Then you can do `response = build_response`